### ISSUE TITLE
GKE - Error getting node pricing

### ISSUE THREAD

**Issue by caueasantos** [opencost/opencost#2091]
**Describe the bug**

It seems that OpenCost is failing to collect pricing from Google Billing API
```yaml
│ opencost 2023-08-15T22:19:25.82045227Z INF Error getting node pricing. Error: Warning: no pricing data found for &{map[addon.gke.io/node-local-dns-ds-ready:true beta.kubernetes.io/arch:amd64 beta.kubernetes.io/instance-type:t2d-stan │
│ dard-8 beta.kubernetes.io/os:linux cloud.google.com/gke-boot-disk:pd-ssd cloud.google.com/gke-container-runtime:containerd cloud.google.com/gke-cpu-scaling-level:8 cloud.google.com/gke-logging-variant:DEFAULT cloud.google.com/gke-ma │
│ x-pods-per-node:110 cloud.google.com/gke-netd-ready:true cloud.google.com/gke-nodepool:small cloud.google.com/gke-os-distribution:cos cloud.google.com/gke-preemptible:true cloud.google.com/gke-provisioning:preemptible cloud.google.c │
│ om/gke-stack-type:IPV4 cloud.google.com/machine-family:t2d cluster_name:admin failure-domain.beta.kubernetes.io/region:us-east1 failure-domain.beta.kubernetes.io/zone:us-east1-c iam.gke.io/gke-metadata-server-enabled:true kubernetes │
│ .io/arch:amd64 kubernetes.io/hostname:gke-admin-small-9bfd2f4e-vf6s kubernetes.io/os:linux node.kubernetes.io/instance-type:t2d-standard-8 node_pool:small node_type:small providerID:gce://<redacted>/us-east1-c/gke-admin-small-9bf │
│ d2f4e-vf6s topology.gke.io/zone:us-east1-c topology.kubernetes.io/region:us-east1 topology.kubernetes.io/zone:us-east1-c]}
```
```yaml
│ opencost 2023-08-15T22:19:25.82050455Z INF Error getting node pricing. Error: Warning: no pricing data found for &{map[addon.gke.io/node-local-dns-ds-ready:true beta.kubernetes.io/arch:amd64 beta.kubernetes.io/instance-type:t2d-stan │
│ dard-8 beta.kubernetes.io/os:linux cloud.google.com/gke-boot-disk:pd-ssd cloud.google.com/gke-container-runtime:containerd cloud.google.com/gke-cpu-scaling-level:8 cloud.google.com/gke-logging-variant:DEFAULT cloud.google.com/gke-ma │
│ x-pods-per-node:110 cloud.google.com/gke-netd-ready:true cloud.google.com/gke-nodepool:static cloud.google.com/gke-os-distribution:cos cloud.google.com/gke-provisioning:standard cloud.google.com/gke-stack-type:IPV4 cloud.google.com/ │
│ machine-family:t2d cluster_name:admin failure-domain.beta.kubernetes.io/region:us-east1 failure-domain.beta.kubernetes.io/zone:us-east1-c iam.gke.io/gke-metadata-server-enabled:true kubernetes.io/arch:amd64 kubernetes.io/hostname:gk │
│ e-admin-static-dedf14bc-ilug kubernetes.io/os:linux node.kubernetes.io/instance-type:t2d-standard-8 node_pool:static node_type:static providerID:gce://<redacted>/us-east1-c/gke-admin-static-dedf14bc-ilug topology.gke.io/zone:us-e │
│ ast1-c topology.kubernetes.io/region:us-east1 topology.kubernetes.io/zone:us-east1-c]}
```
```yaml
│ opencost 2023-08-15T22:19:25.82054173Z INF Error getting node pricing. Error: Warning: no pricing data found for &{map[addon.gke.io/node-local-dns-ds-ready:true beta.kubernetes.io/arch:amd64 beta.kubernetes.io/instance-type:t2d-stan │
│ dard-16 beta.kubernetes.io/os:linux cloud.google.com/gke-boot-disk:pd-ssd cloud.google.com/gke-container-runtime:containerd cloud.google.com/gke-cpu-scaling-level:16 cloud.google.com/gke-logging-variant:DEFAULT cloud.google.com/gke- │
│ max-pods-per-node:110 cloud.google.com/gke-netd-ready:true cloud.google.com/gke-nodepool:large cloud.google.com/gke-os-distribution:cos cloud.google.com/gke-preemptible:true cloud.google.com/gke-provisioning:preemptible cloud.google │
│ .com/gke-stack-type:IPV4 cloud.google.com/machine-family:t2d cluster_name:admin failure-domain.beta.kubernetes.io/region:us-east1 failure-domain.beta.kubernetes.io/zone:us-east1-c iam.gke.io/gke-metadata-server-enabled:true kubernet │
│ es.io/arch:amd64 kubernetes.io/hostname:gke-admin-large-9f140386-hk4x kubernetes.io/os:linux node.kubernetes.io/instance-type:t2d-standard-16 node_pool:large node_type:large providerID:gce://<redacted>/us-east1-c/gke-admin-large- │
│ 9f140386-hk4x topology.gke.io/zone:us-east1-c topology.kubernetes.io/region:us-east1 topology.kubernetes.io/zone:us-east1-c]}
```

**Which version of OpenCost are you using?**

```yaml
      containers:
        - name: opencost
          image: "quay.io/kubecost1/kubecost-cost-model:prod-1.105.0"
        - name: opencost-ui
          image:  "quay.io/kubecost1/opencost-ui:prod-1.105.0"
```

**Which version of Kubernetes are you using?**

GKE Cluster - 1.26.6-gke.1700 

**Expected behaviour**

I'm not sure if that is a bug or if I missed any crucial step to get that working. Can someone please provide some guidance?

**Comment by mattray** [comment]
Do you have any GKE data and the logs have errors or is it not working at all? 
Have you followed https://www.opencost.io/docs/configuration/gcp-opencost?

**Comment by caueasantos** [comment]
I followed that document and also tested the API key locally which seems to have the proper privileges. What kind of GKE logs do you need? 

The OpenCost container is running and I can see some pricing for each namespace within the UI, but I'm not sure if the price numbers are accurate due the errors I shared above.

**Comment by mattray** [comment]
There was a node pricing issue in 1.105.0 and 1.105.1 that was fixed in 1.105.2 with regards to deleted nodes. Were these nodes deleted? You should definitely upgrade to 1.105.2 regardless, but please verify after upgrading.

**Comment by caueasantos** [comment]
Will do!
